tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -0,0 +1,579 @@
+package mixer
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	bookinfoYaml             = "demos/apps/bookinfo/bookinfo.yaml"
+	rulesDir                 = "demos/apps/bookinfo"
+	rateLimitRule            = "mixer-rule-ratings-ratelimit.yaml"
+	denialRule               = "mixer-rule-ratings-denial.yaml"
+	newTelemetryRule         = "mixer-rule-additional-telemetry.yaml"
+	routeAllRule             = "route-rule-all-v1.yaml"
+	routeReviewsVersionsRule = "route-rule-reviews-v2-v3.yaml"
+	emptyRule                = "mixer-rule-empty-rule.yaml"
+
+	mixerAPIPort        = 9094
+	mixerPrometheusPort = 42422
+
+	targetLabel       = "target"
+	responseCodeLabel = "response_code"
+
+	global = "global"
+)
+
+type testConfig struct {
+	*framework.CommonConfig
+	gateway  string
+	rulesDir string
+}
+
+var (
+	tc    *testConfig
+	rules = []string{rateLimitRule, denialRule, newTelemetryRule, routeAllRule, routeReviewsVersionsRule, emptyRule}
+)
+
+func (t *testConfig) Setup() error {
+	t.gateway = "http://" + tc.Kube.Ingress
+	for _, rule := range rules {
+		src := util.GetResourcePath(filepath.Join(rulesDir, fmt.Sprintf("%s", rule)))
+		dest := filepath.Join(t.rulesDir, rule)
+		ori, err := ioutil.ReadFile(src)
+		if err != nil {
+			glog.Errorf("Failed to read original rule file %s", src)
+			return err
+		}
+		content := string(ori)
+		err = ioutil.WriteFile(dest, []byte(content), 0600)
+		if err != nil {
+			glog.Errorf("Failed to write into new rule file %s", dest)
+			return err
+		}
+	}
+	return nil
+}
+func (t *testConfig) Teardown() error { return nil }
+
+type mixerProxy struct {
+	namespace             string
+	metricsPortFwdProcess *os.Process
+	apiPortFwdProcess     *os.Process
+}
+
+func newMixerProxy(namespace string) *mixerProxy {
+	return &mixerProxy{
+		namespace: namespace,
+	}
+}
+
+func (m *mixerProxy) Setup() error {
+	var pod string
+	var err error
+	glog.Info("Setting up mixer proxy")
+
+	getName := fmt.Sprintf("kubectl -n %s get pod -l istio=mixer -o jsonpath='{.items[0].metadata.name}'", m.namespace)
+	pod, err = util.Shell(getName)
+	if err != nil {
+		return err
+	}
+
+	metricsPortFwd := fmt.Sprintf("kubectl port-forward %s %d:%d -n %s", strings.Trim(pod, "'"), mixerPrometheusPort, mixerPrometheusPort, m.namespace)
+	if m.metricsPortFwdProcess, err = util.RunBackground(metricsPortFwd); err != nil {
+		glog.Errorf("Failed to port forward: %s", err)
+		return err
+	}
+	glog.Infof("mixer metrics port-forward running background, pid = %d", m.metricsPortFwdProcess.Pid)
+
+	apiPortFwd := fmt.Sprintf("kubectl port-forward %s %d:%d -n %s", strings.Trim(pod, "'"), mixerAPIPort, mixerAPIPort, m.namespace)
```

Comment:
Liens 123-130 shouldn't be needed anymore with https://github.com/istio/istio/pull/205.